staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -223,3 +226,28 @@ func (f *FieldManager) buildManagerInfo(prefix string, operation metav1.ManagedF
 	}
 	return internal.BuildManagerIdentifier(&managerInfo)
 }
+
+// stripSet is the list of fields that should never be part of a mangedFields.
+var stripSet = fieldpath.NewSet(
+	fieldpath.MakePathOrDie("apiVersion"),
+	fieldpath.MakePathOrDie("kind"),
+	fieldpath.MakePathOrDie("metadata", "name"),
+	fieldpath.MakePathOrDie("metadata", "namespace"),
+	fieldpath.MakePathOrDie("metadata", "creationTimestamp"),
+	fieldpath.MakePathOrDie("metadata", "selfLink"),
+	fieldpath.MakePathOrDie("metadata", "uid"),
+	fieldpath.MakePathOrDie("metadata", "resourceVersion"),
+)
+
+// stripFields removes a predefined set of paths found in typed from managed and returns the updated ManagedFields
+func (f *FieldManager) stripFields(managed fieldpath.ManagedFields, manager string) fieldpath.ManagedFields {
+	vs, ok := managed[manager]
+	if ok && vs != nil {
```

Comment:
I suspect the test on `ok` here is not useful? Also I'm not opposed to panicing if ok is `true` but vs is `nil` (to detect these developers error, though even with the panic it would be hard to find why we have nil here, and we should probably have paniced before).

Reply:
But yeah, I agree that it shouldn't happen in the first place, and so we shouldn't just silently hide a bug.